plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSourceProvider.java

```diff
@@ -385,6 +391,26 @@ private static boolean shouldSkipBucket(HiveTableHandle hiveTable, HiveSplit hiv
         return hiveBucketFilter.map(filter -> !filter.getBucketsToKeep().contains(hiveSplit.getBucketNumber().getAsInt())).orElse(false);
     }
 
+    private static boolean shouldSkipSplit(List<ColumnMapping> columnMappings, DynamicFilter dynamicFilter)
+    {
+        TupleDomain<ColumnHandle> predicate = dynamicFilter.getCurrentPredicate();
```

Comment:
It seems we're considering DF filter over and over again.
Add code-level comment (or at least commit message) why this place is useful even though we have other places.